internal/registry/registry.go

```diff
@@ -263,7 +263,7 @@ func (r *Registry) Register(ctx context.Context, info *ProviderInfo) error {
 		return err
 	}
 
-	allowed, trusted := r.policy.Check(info.AddrInfo.ID)
+	allowed, trusted, _ := r.policy.Check(info.AddrInfo.ID)
```

Comment:
Trusted now means not rate-limited, the below on line 276 would need to change.  Do not want to deny access because of rate limiting.